src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -23,17 +24,25 @@ namespace Microsoft.Health.Events.EventCheckpointing
 {
     public class StorageCheckpointClient : ICheckpointClient
     {
-        private ConcurrentDictionary<string, Checkpoint> _checkpoints;
-        private ConcurrentDictionary<string, int> _lastCheckpointTracker;
-        private int _lastCheckpointMaxCount;
-        private BlobContainerClient _storageClient;
-        private ITelemetryLogger _log;
-
-        public StorageCheckpointClient(BlobContainerClient containerClient, StorageCheckpointOptions options, ITelemetryLogger log)
+        private readonly string _blobCheckpointPrefix;
+        private readonly string _blobPath;
+        private readonly ConcurrentDictionary<string, Checkpoint> _checkpoints;
+        private readonly int _lastCheckpointMaxCount;
+        private readonly ConcurrentDictionary<string, int> _lastCheckpointTracker;
+        private readonly ITelemetryLogger _log;
+        private readonly BlobContainerClient _storageClient;
+
+        public StorageCheckpointClient(BlobContainerClient containerClient, StorageCheckpointOptions options, EventHubClientOptions eventHubClientOptions, ITelemetryLogger log)
         {
             EnsureArg.IsNotNull(containerClient);
             EnsureArg.IsNotNull(options);
-            BlobPrefix = options.BlobPrefix;
+            EnsureArg.IsNotNull(eventHubClientOptions);
+
+            // Blob path for checkpoints includes the event hub name to scope the checkpoints per source event hub.
+            _blobCheckpointPrefix = $"{options.BlobPrefix}/checkpoint/";
+            _blobPath = eventHubClientOptions != default ?
+                $"{_blobCheckpointPrefix}{eventHubClientOptions.EventHubNamespaceFQDN}/{eventHubClientOptions.EventHubName}/" :
+                _blobCheckpointPrefix;
```

Comment:
EventHubNamespaceFQDN isn't required to be set in all cases. When using a connection string, the user only has to specify a connection string and the consumer group. 

If the user does not specify EventHubNamespaceFQDN, then the blob path looking like this:
checkpoint/Normalization//checkpoint//devicedata

We can still make EventHubNamespaceFQDN required if we want. Just wanted to point out that it is not currently required by the EventProcessorClient constructor when using a connection string. EventHubNamespaceFQDN  is only required by the EventProcessorClient constructor when using managed identity.

[EventProcessorClientFactory](https://github.com/microsoft/iomt-fhir/blob/7794cbcc463e8d26c3097cd5e2243d770f26fe45/src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessorClientFactory.cs)

Reply:
Updated to check the connection string properties if that is the auth type selected. 
(Thanks for bringing it up, didn't cross my mind, since I was using the default appsettings.json which had the FQDN and eventhub name params in there!)